sql/src/main/java/io/crate/execution/engine/sort/SortingTopNCollector.java

```diff
@@ -79,6 +80,9 @@ public SortingTopNCollector(Collection<? extends Input<?>> inputs,
         this.numOutputs = numOutputs;
         this.comparator = comparator;
         this.offset = offset;
+        if (numExpectedRows > 0) {
+            limit = Math.min(((Long) numExpectedRows).intValue(), limit);
```

Comment:
We can't use an estimate that was determined in the planner phase as an implicit limit replacement. This will lead to incorrent results. 

We'll have to come up with a different approach that takes into account the *actual* number of source rows.

Reply:
Yes, I thought that might be the case *sigh. I believe we currently don't have this information though, especially as results come in pages